test/CMakeLists.txt

```diff
@@ -58,6 +58,8 @@ doctest_discover_tests(test-encodings)
 
 add_executable(test-swiftnet doctest_swiftnet.cpp)
 add_sycl_to_target(TARGET test-swiftnet SOURCES doctest_swiftnet.cpp)
+target_compile_definitions(test-swiftnet PRIVATE TEST_GRAPH)
```

Comment:
Only used in trainer. Can we remove this entirely and just always use the graph version?